lib/bolt/boltdir.rb

```diff
@@ -12,13 +12,21 @@ def self.default_boltdir
       Boltdir.new(File.join('~', '.puppetlabs', 'bolt'))
     end
 
+    # Search recursively up the directory hierarchy for the Boltdir. Look for a
+    # directory called Boltdir or a file called bolt.yaml (for a control repo
+    # type Boltdir). Otherwise, repeat the check on each directory up the
+    # hierarchy, falling back to the default if we reach the root.
     def self.find_boltdir(dir)
-      local_boltdir = Pathname.new(dir).ascend do |path|
-        boltdir = path + BOLTDIR_NAME
-        break new(boltdir) if boltdir.directory?
+      dir = Pathname.new(dir)
+      if (dir + BOLTDIR_NAME).directory?
+        new(dir + BOLTDIR_NAME)
+      elsif (dir + 'bolt.yaml').file?
+        new(dir)
+      elsif dir.root?
```

Comment:
lets make sure we have a good windows test for finding the default

Reply:
Added an explicit test of that.